elastic/kibana
# [Cases] Case action

## Summary

Depends on: https://github.com/elastic/kibana/pull/166267, https://github.com/elastic/kibana/pull/170326, https://github.com/elastic/kibana/pull/169484, https://github.com/elastic/kibana/pull/173740, https://github.com/elastic/kibana/pull/173763, https://github.com/elastic/kibana/pull/178068, https://github.com/elastic/kibana/pull/178307, https://github.com/elastic/kibana/pull/178600, https://github.com/elastic/kibana/pull/180437

PRs:
- https://github.com/elastic/kibana/pull/168370
- https://github.com/elastic/kibana/pull/169229
- https://github.com/elastic/kibana/pull/171754
- https://github.com/elastic/kibana/pull/172709
- https://github.com/elastic/kibana/pull/173012
- https://github.com/elastic/kibana/pull/175107
- https://github.com/elastic/kibana/pull/175452
- https://github.com/elastic/kibana/pull/175505
- https://github.com/elastic/kibana/pull/177033
- https://github.com/elastic/kibana/pull/178277
- https://github.com/elastic/kibana/pull/177139
- https://github.com/elastic/kibana/pull/179796

Fixes: https://github.com/elastic/kibana/issues/153837

## Testing

Run Kibana with `--run-examples` if you want to use the "Always firing" rule.

Create a rule with a case action in observability and the stack. The security solution is not supported. You should not be able to assign a case action in a security solution rule.

1. Test the "Reopen closed cases" configuration.
2. Test the "Grouping by" configuration. Only one field is allowed. Not all fields are persisted in alerts. If you select a field not part of the alert the case action will create a case where the grouping value is set to `unknow`.
3. Test the "Time window" feature. You can comment out the validation to test for shorter times.
4. Verify that the case action is experimental.
5. Verify that based on the rule type the case is created in the correct solution.
6. Verify that you cannot create a rule with the case action on the basic license.
7. Verify that the execution of the case action fails if you do not have permission for cases. Pending work on the system actions framework level to not allow users to create rules with system actions where they do not have permission.
8. Stress test the case action by creating multiple rules.

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers


## Release notes

Automatically create cases when an alert is triggered.
